### PR TITLE
Add `pin_logo` `is_touched`.

### PR DESCRIPTION
**Pull request by microbit-robert**

### CODE REVIEW

File: src/board.js [hunk 1]
```diff
@@ -191,6 +191,9 @@ class BoardUI {
       new ButtonUI(this.svg.querySelector("#ButtonA"), "A"),
       new ButtonUI(this.svg.querySelector("#ButtonB"), "B"),
     ];
+    this.pins = Array(33)
+    this.pins[MICROBIT_HAL_PIN_FACE] = 
+      new PinUI(this.svg.querySelector("#Logo"), "pin_logo"),
```
Comment (microbit-robert) [inline src/board.js:196]: I suspect this needs improving. Looks inside an array by index. `MICROBIT_HAL_PIN_FACE = 19`.

File: src/board.js [hunk 2]
```diff
@@ -483,3 +488,72 @@ function clamp(value, min, max) {
...
+    this._isTouched = false;
+    this.render();
+  }
+
+  isTouched() {
+    return this._isTouched;
+  }
+
+  render() {
+    const fill = this._isTouched ? "red" : "url(#an)";
```
Comment (microbit-robert) [inline src/board.js:550]: More appropriate colour than red?

Comment (microbit-matt-hillsdon) [reply src/board.js:550]: I think we can get this in and discuss the UI of the buttons/logo later. The colour choice for the buttons was similarly arbitrary. MakeCode fill the background of the logo which is a bit misleading but may well have been discussed at length.

File: src/board.js (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply src/board.js:196]: I think this is fine for now

File: src/demo.html [hunk 3]
```diff
@@ -78,6 +78,7 @@ <h1>MicroPython-micro:bit simulator example embedding</h1>
               <option value="display">Display</option>
               <option value="music">Music</option>
               <option value="sensors">Sensors</option>
+              <option value="pin_logo">Pin logo</option>
```
Comment (microbit-matt-hillsdon) [inline src/demo.html:80]: Please can we keep these alphabetical and amend the comment to say they should be? Aiming to reduce trivial conflicts.

Comment (microbit-robert) [reply src/demo.html:80]: [`493e35f`](https://github.com/microbit-foundation/micropython-microbit-v2-simulator/pull/4/commits/493e35fb61319d0b0050bd5bb78e6e6469407872).

**Review by microbit-matt-hillsdon — APPROVED** [review]
Looks good. We'll discuss the UI later.

**Pull request merged**